BudgetTracker.Business/Budgeting/Budget.cs

```diff
@@ -196,6 +191,31 @@ public async Task<Budget> GetRootBudget(IBudgetRepository budgetRepository)
         }
         private Budget _rootBudget;
 
+        /// <summary>
+        /// <p>
+        /// Fetches all transactions that have been logged under this budget
+        /// or one of it's sub budgets. This is recursive. This budgets sub
+        /// budgets must be loaded recursively before calling this method.
+        /// </p>
+        /// </summary>
+        public async Task<List<Transaction>> GetTransactions(DateTime? fromDate, DateTime? toDate, ITransactionRepository transactionRepository)

```

Comment:
Its best practice to use IEnumerable<> instead of LIst<> 

https://stackoverflow.com/questions/3628425/ienumerable-vs-list-what-to-use-how-do-they-work